plugins/catalog-backend-module-aws-accounts/src/providers/AwsOrganizationProvider.ts

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  EntityProvider,
+  EntityProviderConnection,
+  DeferredEntity,
+} from '@backstage/plugin-catalog-node';
+import * as uuid from 'uuid';
+import {
+  LoggerService,
+  SchedulerService,
+  SchedulerServiceTaskRunner,
+} from '@backstage/backend-plugin-api';
+import {
+  AwsCredentialProvider,
+  AwsCredentialsManager,
+  DefaultAwsCredentialsManager,
+} from '@backstage/integration-aws-node';
+import {
+  Account,
+  Organizations,
+  paginateListAccounts,
+} from '@aws-sdk/client-organizations';
+import { ResourceEntityV1alpha1 } from '@backstage/catalog-model';
+import { Config } from '@backstage/config';
+import {
+  readProviderConfigs,
+  AwsOrganizationEntityProviderConfig,
+} from '../config';
+
+const AWS_ORGANIZATION_REGION = 'us-east-1';
+const ACCOUNT_STATUS_LABEL = 'amazonaws.com/account-status';
+const ACCOUNTID_ANNOTATION = 'amazonaws.com/account-id';
+const ACCOUNT_EMAIL_ANNOTATION = 'amazonaws.com/account-email';
+const ARN_ANNOTATION = 'amazonaws.com/arn';
+const ORGANIZATION_ANNOTATION = 'amazonaws.com/organization-id';
+
+/**
+ * Catalog provider to ingest AWS Organization accounts.
+ *
+ * @public
+ */
+export class AwsOrganizationEntityProvider implements EntityProvider {
+  private readonly providerConfig: AwsOrganizationEntityProviderConfig;
+  private readonly logger: LoggerService;
+  private readonly scheduleFn: () => Promise<void>;
+  private readonly organizations: Organizations;
+  private connection?: EntityProviderConnection;
+
+  private constructor(
+    providerConfig: AwsOrganizationEntityProviderConfig,
+    logger: LoggerService,
+    taskRunner: SchedulerServiceTaskRunner,
+    private readonly credProvider: AwsCredentialProvider,
+  ) {
+    this.providerConfig = providerConfig;
+    this.logger = logger.child({
+      target: this.getProviderName(),
+    });
+    this.scheduleFn = this.createScheduleFn(taskRunner);
+    this.organizations = new Organizations({
+      customUserAgent: 'backstage-aws-catalog-organizations',
+      credentialDefaultProvider: () => this.credProvider.sdkCredentialProvider,
+      region: AWS_ORGANIZATION_REGION,
+    }); // Only available in us-east-1
+  }
+
+  getProviderName(): string {
+    return `aws-org-provider:${this.providerConfig.id}`;
```

Comment:
same here, is it the accounts provider or the org provider? :)

honestly, most providers that produce users and groups are called org providers, so that one might be the most sensible - but the work to rename the entire module folder etc is a bit bigger - your call